planner/core/point_get_plan.go

```diff
@@ -1039,6 +1066,22 @@ func getNameValuePairs(stmtCtx *stmtctx.StatementContext, tbl *model.TableInfo,
 	return nil, false
 }
 
+func checkCanUseConvertTo(col *model.ColumnInfo, d types.Datum) bool {
+	dKind := d.Kind()
+	switch col.FieldType.EvalType() {
+	case ptypes.ETString:
+		switch dKind {
```

Comment:
Should we use the white list?

Reply:
For now I just find String type column with numeric constant will cause the bug. So I just use black list style.